gallery/experiments/experimental_abinitio_pipeline_10073_jsb.py

```diff
@@ -0,0 +1,167 @@
+n_imgs = None  # Set to None for all images in starfile, can set smaller for tests.
+img_size = 129  # Downsample the images/reconstruction to a desired resolution
+n_classes = 3000  # How many class averages to compute.
+n_nbor = 50  # How many neighbors to stack
+
+# Outputs
+preprocessed_fn = f"10073_preprocessed_{img_size}px.star"
+class_avg_fn = f"10073_var_sorted_cls_avgs_m{n_nbor}_{img_size}px.star"
+oriented_fn = f"10073_oriented_class_averages_{img_size}px.star"
+volume_output_filename = f"10073_abinitio_c{n_classes}_m{n_nbor}_{img_size}px.mrc"
+
+
+# %%
+# Source data and Preprocessing
+# -----------------------------
+#
+# ``RelionSource`` is used to access the experimental data via a `STAR` file.
+
+# Create a source object for the experimental images.
+src = RelionSource(
+    starfile_in, pixel_size=pixel_size, max_rows=n_imgs, data_folder=data_folder
+)
+
+# Use ``phase_flip`` to attempt correcting for CTF.
+logger.info("Perform phase flip to input images.")
+src = src.phase_flip().cache()
+
+# Downsample the images.
+logger.info(f"Set the resolution to {img_size} X {img_size}")
+src = src.downsample(img_size).cache()
+
+# Normalize the background of the images.
+src = src.normalize_background().cache()
+
+# Estimate the noise and whiten based on the estimated noise.
+src = src.legacy_whiten().cache()
+
+# Optionally invert image contrast.
+logger.info("Invert the global density contrast")
+src = src.invert_contrast().cache()
+
+# Save the preprocessed image stack.
+src.save(preprocessed_fn, save_mode="single", overwrite=True)
+
+
+# %%
+# Class Averaging
+# ----------------------
+#
+# Now perform classification and averaging for each class.
+
+logger.info("Begin Class Averaging")
+
+avgs = LegacyClassAvgSource(src, n_nbor=n_nbor).cache()
+
+# Save the entire set of class averages to disk so they can be reused.
+avgs.save(class_avg_fn, save_mode="single", overwrite=True)
+
+# We'll continue our pipeline by selecting``n_classes`` from ``avgs``.
+# To capture a broader range of viewing angles, uniformly select every ``k`` image.
+k = (avgs.n - 1) // n_classes
+avgs = avgs[::k].cache()
+
+
+# %%
+# Common Line Estimation
+# ----------------------
+#
+# Estimate orientation of projections and assign to source by
+# applying ``OrientedSource`` to the class averages from the prior
+# step. By default this applies the Common Line with Synchronization
+# Voting ``CLSync3N`` method.  Here additional weighting techniques
+# are applied for common lines detection by customizing the
+# orientation estimator component.
+
+logger.info("Apply custom mask")
+# 10073 benefits from a masking procedure that is more aggressive than the default.
+# Note, since we've manually masked, the default masking is disabled below in `CLSync3N`.
+# This also upcasts to double precision, which is helpful for this reconstruction.
+mask = fuzzy_mask((img_size, img_size), np.float64, r0=0.4 * img_size, risetime=2)
+avgs = ArrayImageSource(avgs.images[:] * mask)
```

Comment:
This is not a great look. I guess we should look into making this a preprocessing method?

Reply:
Agreed, but for a custom application I don't think its that bad.

Its only a minor improvement.  Actually, the FSC come out virtually the same, but the appearance is better using the mask.  I considered removing it, but I fear that when other components are improved bringing the recons closer to the paper result, it might turn out to be more important than it is now...

Reply:
I'm not I think it is worth building out code for another preprocessing method.  It is unlikely it would save a user any lines of code.

Reply:
All right.